corehq/apps/sso/static/sso/js/enterprise_edit_identity_provider.js

```diff
@@ -55,6 +56,39 @@ hqDefine('sso/js/enterprise_edit_identity_provider', [
 
         };
 
+        let remoteUserManagementAPISecretManager = function () {
+            let self = {};
+            self.isCancelUpdateVisible = ko.observable(false);
+            self.apiExpirationDate = "";
+
+
+            self.isAPISecretVisible =  ko.observable(!initialPageData.get('api_secret_exists'));
+
+            self.showAPISecret = function () {
+                self.isAPISecretVisible(true);
+                self.isCancelUpdateVisible(true);
+                // Store the current expiration date before clearing them for editing.
+                self.apiExpirationDate = document.getElementById('id_date_api_secret_expiration').value;
+                document.getElementById('id_date_api_secret_expiration').value = '';
+            };
+
+            self.hideAPISecret = function () {
+                self.isAPISecretVisible(false);
+                self.isCancelUpdateVisible(false);
+                document.getElementById('id_api_secret').value = '';
```

Comment:
it's a bit strange to mix the `document.getElementById(<selector>)` functionality when the page is both using knockout and jQuery.

It is preferable to bind a knockout observable to the relevant fields you want to get/set in the knockout code, rather than doing it manually like this. it can confuse the reader and doesn't make the best use out of knockout.